src/steps/user-groups/index.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import {
+  ACCOUNT_ENTITY_KEY,
+  Entities,
+  Relationships,
+  Steps,
+} from '../constants';
+import { createUserGroupEntity } from './converter';
+
+export async function fetchUserGroups({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateUserGroups(async (userGroup) => {
+    const userGroupEntity = createUserGroupEntity(userGroup);
+    if (userGroupEntity) {
+      await jobState.addEntity(userGroupEntity);
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.HAS,
+          from: (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity,
+          to: userGroupEntity,
+        }),
+      );
+    }
+  });
+}
```

Comment:
I'd put `await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity` somewhere at the beginning, that we way we only have to fetch it one time for arbitrary number of user groups there may be. Another benefit is we can check if we have access to it before attempting to create relationship.

```suggestion
  await apiClient.iterateUserGroups(async (userGroup) => {
    const accountEntity = await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity;
  
    const userGroupEntity = createUserGroupEntity(userGroup);
    if (userGroupEntity) {
      await jobState.addEntity(userGroupEntity);

      if (accountEntity) {
        await jobState.addRelationship(
          createDirectRelationship({
            _class: RelationshipClass.HAS,
            from: accountEntity,
            to: userGroupEntity,
          }),
        );
      }
    }
  });
}
```